system-addon/lib/TippyTopProvider.jsm

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const {utils: Cu} = Components;
+
+Cu.importGlobalProperties(["fetch", "URL"]);
+
+const TIPPYTOP_JSON_PATH = "resource://activity-stream/data/content/tippytop/top_sites.json";
+const TIPPYTOP_URL_PREFIX = "resource://activity-stream/data/content/tippytop/images/";
+
+function getDomain(url) {
+  let domain = new URL(url).hostname;
+  if (domain && domain.startsWith("www.")) {
+    domain = domain.slice(4);
+  }
+  return domain;
+}
+
+function getPath(url) {
+  return new URL(url).pathname;
+}
+
+this.TippyTopProvider = class TippyTopProvider {
+  constructor() {
+    this._sitesByDomain = {};
+  }
+  async init() {
+    // Load the Tippy Top sites from the json manifest.
+    fetch(TIPPYTOP_JSON_PATH)
+      .then(response => response.json())
+      .then(sites => {
```

Comment:
Do I need to stub it out differently in tests to do this? I can't get them to work now for some reason.